corehq/apps/reports/management/commands/create_tableau_connection.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from django.core.management import BaseCommand
+
+from corehq.apps.reports.models import TableauAppConnection, TableauDomainDetails
+
+logger = logging.getLogger('create_tableau_connection')
+logger.setLevel('DEBUG')
+
+
+class Command(BaseCommand):
+    help = ''' This command will associate a given set of domains with a Tableau connected app, enabling the use
+              of Embedded Tableau on those domains. You can learn about creating a connected app here:
+              https://help.tableau.com/current/server/en-us/connected_apps.htm#create-a-connected-app.'''
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--domains',
+            nargs='+',
+            required=True,
+            help="A list of domains to associate with the inputted connected app.",
+        )
+        parser.add_argument(
+            '--server-name',
+            required=True,
+            help="The host name of the server that is running Tableau.",
+        )
+        parser.add_argument(
+            '--site-name',
+            required=True,
+            help="The Tableau site you wish to connect your domain(s) with.",
+        )
+        parser.add_argument(
+            '--app-client-id',
+            required=True,
+            help="The unique ID for the connected app."
+        )
+
+    def prompt_for_secret_details(self):
+        self.secret_id = input('Please enter in the ID of the secret for the connected app:\n')
+        self.secret_value = input('Please enter in the value for this secret:\n')
```

Comment:
nit: I'd return these two values rather than setting them as instance attributes.  That would make the dependency between this method and the places these are used more explicit, and would make the lifecycle of these values more obvious.